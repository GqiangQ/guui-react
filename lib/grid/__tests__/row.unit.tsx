import {configure} from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
import * as React from 'react';
import * as renderer from 'react-test-renderer';
import Row from '../row';

configure({adapter: new Adapter()});

describe(Row.displayName, () => {
  it('DisplayName', () => {
    expect(Row.displayName).toEqual('Row');
  });

  describe('className', () => {
    it(' = xxx', () => {
      const tree = renderer
        .create(<Row className="xxx"></Row>)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  describe('style', () => {
    it(` = {color: 'red'}`, () => {
      const tree = renderer
        .create(<Row style={{color: 'red'}}></Row>)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  describe('gutter', () => {
    it(` = 16`, () => {
      const tree = renderer
        .create(<Row gutter={16}></Row>)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  describe('align', () => {
    it(` = right`, () => {
      const tree = renderer
        .create(<Row align="right"></Row>)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
    it(` = center`, () => {
      const tree = renderer
        .create(<Row align="center"></Row>)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
    it(` = spaceBetween`, () => {
      const tree = renderer
        .create(<Row align="spaceBetween"></Row>)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
    it(` = spaceAround`, () => {
      const tree = renderer
        .create(<Row align="spaceAround"></Row>)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  describe('verticalAlign', () => {
    it(`= center`, () => {
      const tree = renderer
        .create(<Row verticalAlign="center"></Row>)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
    it(`= bottom`, () => {
      const tree = renderer
        .create(<Row verticalAlign="bottom"></Row>)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
