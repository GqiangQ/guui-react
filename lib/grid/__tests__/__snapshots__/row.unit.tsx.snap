// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row align  = center 1`] = `
<div
  className="gu-row gu-row-alignCenter "
  style={
    Object {
      "margin": undefined,
    }
  }
/>
`;

exports[`Row align  = right 1`] = `
<div
  className="gu-row gu-row-alignRight "
  style={
    Object {
      "margin": undefined,
    }
  }
/>
`;

exports[`Row align  = spaceAround 1`] = `
<div
  className="gu-row gu-row-alignSpaceAround "
  style={
    Object {
      "margin": undefined,
    }
  }
/>
`;

exports[`Row align  = spaceBetween 1`] = `
<div
  className="gu-row gu-row-alignSpaceBetween "
  style={
    Object {
      "margin": undefined,
    }
  }
/>
`;

exports[`Row className  = xxx 1`] = `
<div
  className="gu-row gu-row-alignLeft xxx"
  style={
    Object {
      "margin": undefined,
    }
  }
/>
`;

exports[`Row gutter  = 16 1`] = `
<div
  className="gu-row gu-row-alignLeft "
  style={
    Object {
      "margin": "0 -8px",
    }
  }
/>
`;

exports[`Row style  = {color: 'red'} 1`] = `
<div
  className="gu-row gu-row-alignLeft "
  style={
    Object {
      "color": "red",
      "margin": undefined,
    }
  }
/>
`;

exports[`Row verticalAlign = bottom 1`] = `
<div
  className="gu-row gu-row-alignLeft gu-row-verticalAlignBottom "
  style={
    Object {
      "margin": undefined,
    }
  }
/>
`;

exports[`Row verticalAlign = center 1`] = `
<div
  className="gu-row gu-row-alignLeft gu-row-verticalAlignCenter "
  style={
    Object {
      "margin": undefined,
    }
  }
/>
`;
